### ISSUE TITLE
Attach a HMC sampler to get full uncertainties

### ISSUE THREAD

**Issue by pmelchior** [pmelchior/scarlet2#3]
With gradients computed automatically, we should provide an option to run a sampler. It probably needs to accept pytrees as parameters, but we may be able to collapse these to standard parameter lists. I've had some trouble getting `blackjax` to run, but here's what I've got:

```python
import blackjax
rng_key = random.PRNGKey(0)
logdensity_fn = lambda x: -loss_fn(x, data)
init_state = eqx.filter(source, eqx.is_array)

adapt = blackjax.window_adaptation(
   blackjax.nuts, logdensity_fn, initial_step_size=0.01
)
last_state, kernel, _ = adapt.run(rng_key, init_state)

# adapt = blackjax.meads(logdensity_fn, 1)
# last_state, kernel, _ = adapt.run(rng_key, init_state)

print(last_state.position.spectrum(), loss_fn(last_state.position, data))

def inference_loop(rng_key, kernel, initial_state, num_samples):
    @jax.jit
    def one_step(state, rng_key):
        state, info = kernel(rng_key, state)
        return state, (state, info)

    keys = jax.random.split(rng_key, num_samples)
    _, (states, infos) = jax.lax.scan(one_step, initial_state, keys)

    return states, infos

num_sample = 1000
states, infos = inference_loop(rng_key, kernel, last_state, num_sample)

print(states.potential_energy[-10:])
print(states.position.spectrum()[-10:])
```

One of the problems is that the window adaptation does not seem to work, and without it the inference is just a mess.
